static/js/message_list_view.js

```diff
@@ -1186,6 +1186,11 @@ export class MessageListView {
         this._rerender_message(message_container, false, true);
     }
 
+    hide_revealed_message(message_id) {
+        const message_container = this.message_containers.get(message_id);
+        this._rerender_message(message_container, false, false);
+    }
```

Comment:
This isn't related to this PR, but maybe it would be more readable if we change `_rerender_message` to take in a dictionary rather than three parameters like this. What do you think?

Reply:
I tam not sure if that would give any benefits also while changing this I think it would be desirable to retain the IDE support(being able to see what variable it takes by hovering over its name). For that it would have to be changed like:
```javascript
_rerender_message({message_container, message_content_edited, is_revealed})
```
so that on hovering one can know what are the properties it will use/extract. So we again have to write each of these separately. That's why I am little doubtful if this would be a useful change.
Although, I may be missing some other aspects so I think @timabbott could have better thoughts on this.

Reply:
My hope was to make *calling* `rerender_message` more cleaner. So, with your example, we will call `rerender_message` like the following, right?
```js
this._rerender_message({
    message_container,
    message_content_edited: false,
    is_revealed: false,
});
```

This I think is more readable than just passing boolean value, which we do now. But yeah, let's wait for more comments.

Reply:
Yeah this looks more clear about what parameters are being passed.

Reply:
Yeah, it'd be well worth doing that refactor in this PR while we're doing manual testing for this feature; I had the same experience of feeling like I needed to open the file to check that those parameters do what I think they do, which would be solved with proper naming.

Reply:
Added [d20fdb7](https://github.com/zulip/zulip/pull/18497/commits/d20fdb7eec1cf3bbbe9894a25e5fe36daff898f1) for this refactoring.